### PR TITLE
Update docs on retries to retry POST requests

### PR DESCRIPTION
**Pull request by TomAugspurger**
This updates the usage docs to set `allowed_methods=False`. The motivation is to retry on `POST` requests to `search` and pagination methods (IIUC, POST is used when `intersects` is provided, since it goes in the request body). By default, requests / urllib3 only retry on

    {'DELETE', 'GET', 'HEAD', 'OPTIONS', 'PUT', 'TRACE'}

https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.Retry.DEFAULT_ALLOWED_METHODS

### CODE REVIEW

**Comment by TomAugspurger** [comment]
CI failure is from https://github.com/stac-utils/pystac-client/issues/562.

The doc build did pass: https://github.com/stac-utils/pystac-client/actions/runs/5570293343/jobs/10174236093?pr=561. OK to merge?

**Review by jsignell — APPROVED** [review]
Yeah LGTM. But it doesn't look like I have quite the right permissions for my approval to count hahaha

File: docs/usage.rst [hunk 1]
```diff
@@ -131,7 +131,9 @@ If you'd like to configure this behavior, e.g. to retry on some ``50x`` response
     from pystac_client import Client
     from pystac_client.stac_api_io import StacApiIO
 
-    retry = Retry(total=5, backoff_factor=1, status_forcelist=[502, 503, 504])
+    retry = Retry(
+        total=5, backoff_factor=1, status_forcelist=[502, 503, 504], allowed_methods=False
```
Comment (Geosynopsis) [inline docs/usage.rst:135]: **suggestion:** ‏I'd recommend including a `frozenset` with `POST` additional to DEFAULT_ALLOWED_METHODS as input to be more explicit. If not, `allowed_methods=None` as recommended by the documentation.

Comment (gadomski) [inline docs/usage.rst:135]:
```suggestion
        total=5, backoff_factor=1, status_forcelist=[502, 503, 504], allowed_methods=None
```

**Review by gadomski — APPROVED** [review]

**Pull request merged**